Resources/public/js/Sonata.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Sonata core object.
+ * Config, translation, and DOM helpers.
+ */
+;(function (window, $, Sonata) {
+    'use strict';
+
+    // Merge default config with data passed from the templates.
+    Sonata.config = $.extend({
+        confirmExit: true,
+        useSelect2: true,
+        useICheck: true,
+        useStickyForms: true
+    }, Sonata.config || {});
+
+    // Merge default translations with data passed from the templates.
+    Sonata.i18n = $.extend({
+        confirmExit: 'You have unsaved changes. Do you really want to leave ?',
+        loadingInformation: 'Loading...'
+    }, Sonata.i18n || {});
+
+    Sonata.Admin = {
+
+        //
+        // Form helpers
+        // ------------------------------------------------------------------------------------------------------------
+
+        /**
+         * Returns the form field with the given id.
+         *
+         * @param {string} fieldId
+         * @param {(HTMLElement|jQuery)} [context]
+         * @returns {jQuery}
+         */
+        getField: function (fieldId, context) {
+            return $('#' + fieldId, context || document);
+        },
+
+        getFormGroup: function (fieldId, context) {
+            return $('#sonata-ba-field-container-' + fieldId, context || document);
+        },
+
+        //
+        // Form association fields helpers
+        // ------------------------------------------------------------------------------------------------------------
+
+        /**
+         * Returns the field container of the association field identified by the given id.
+         * Also has the class .field-container
+         *
+         * This container typically contains:
+         *   * a #field_widget_{id} element containing the widget used to represent the association
+         *   * a #field_actions_{id} element, hosting the action buttons (list, add, delete)
+         *
+         *
+         * @param {string} fieldId
+         * @param {(HTMLElement|jQuery)} [context]
+         * @returns {jQuery}
+         */
+        getAssociationFieldContainer: function (fieldId, context) {
+            return $('#field_container_' + fieldId, context || document);
+        },
```

Comment:
Example of bad naming: there's `#sonata-ba-field-container-*` for standard fields, then `#field_container_*` for association fields... :confused: